2.4/root/usr/share/container-scripts/mongodb/common.sh

```diff
@@ -117,23 +117,60 @@ function mongo_initiate() {
 
 # get the address of the current primary member
 function mongo_primary_member_addr() {
-  local current_endpoints=$(endpoints)
-  local mongo_node="$(echo "${current_endpoints}" | grep -v "$(container_addr)" | head -1):${CONTAINER_PORT}"
-  echo -n $(mongo admin -u admin -p "${MONGODB_ADMIN_PASSWORD}" --host ${mongo_node} --quiet --eval "print(rs.isMaster().primary);")
+  endpoints | grep -v "$(container_addr)" | while read mongo_node; do
+    cmd_output="$(mongo admin -u admin -p "$MONGODB_ADMIN_PASSWORD" --host "$mongo_node:$CONTAINER_PORT" --eval 'print(rs.isMaster().primary)' --quiet || true)"
```

Comment:
Your example is not the same as our code. Compare the following:

```bash
if cmd1 | cmd2 | cm3; then
   echo "no error in the pipe line"
else
   echo "ops, something went bad"
fi
```
and
```bash
out=$(cmd1 | cmd2)
if echo $out | cm3; then
   echo "no error in the pipe line"
else
   echo "ops, something went bad: $out"
fi
```
I believe that we can't inline the `cmd1 | cmd2` here.